src/_pytest/runner.py

```diff
@@ -512,22 +515,21 @@ def teardown_exact(self, nextitem: Optional[Item]) -> None:
         stack is torn down.
         """
         needed_collectors = nextitem and nextitem.listchain() or []
-        exc = None
+        exceptions = []
         while self.stack:
             if list(self.stack.keys()) == needed_collectors[: len(self.stack)]:
                 break
-            node, (finalizers, _) = self.stack.popitem()
+            _, (finalizers, _) = self.stack.popitem()
             while finalizers:
                 fin = finalizers.pop()
                 try:
                     fin()
                 except TEST_OUTCOME as e:
-                    # XXX Only first exception will be seen by user,
-                    #     ideally all should be reported.
-                    if exc is None:
-                        exc = e
-        if exc:
-            raise exc
+                    exceptions.append(e)
+        if len(exceptions) == 1:
+            raise exceptions[0]
+        elif exceptions:
+            raise BaseExceptionGroup("", exceptions)
```

Comment:
Should we add some context here?

```suggestion
            raise BaseExceptionGroup("errors during test teardown", exceptions)
```